edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/ModuleClient.cs

```diff
@@ -2,249 +2,207 @@
+        };
 
-        readonly Client.ModuleClient moduleClient;
+        readonly AtomicBoolean isActive = new AtomicBoolean(true);
+        readonly ISdkModuleClient inner;
+        readonly ResettableTimer inactivityTimer;
 
-        ModuleClient(Client.ModuleClient moduleClient)
+        public ModuleClient(ISdkModuleClient inner, TimeSpan idleTimeout, bool closeOnIdleTimeout)
         {
-            this.moduleClient = moduleClient;
+            this.inner = Preconditions.CheckNotNull(inner, nameof(inner));
+            this.inactivityTimer = new ResettableTimer(this.CloseOnInactivity, idleTimeout, Events.Log, closeOnIdleTimeout);
+            this.inactivityTimer.Start();
         }
 
-        public static Task<IModuleClient> Create(
-            Option<string> connectionString,
-            Option<UpstreamProtocol> upstreamProtocol,
-            ConnectionStatusChangesHandler statusChangedHandler,
-            Func<ModuleClient, Task> initialize,
-            Option<IWebProxy> proxy,
-            Option<string> productInfo) =>
-            Create(upstreamProtocol, initialize, u => CreateAndOpenDeviceClient(u, connectionString, statusChangedHandler, proxy, productInfo));
-
-        public Task SetDesiredPropertyUpdateCallbackAsync(DesiredPropertyUpdateCallback onDesiredPropertyChanged) =>
-            this.moduleClient.SetDesiredPropertyUpdateCallbackAsync(onDesiredPropertyChanged, null);
-
-        public Task SetMethodHandlerAsync(string methodName, MethodCallback callback) =>
-            this.moduleClient.SetMethodHandlerAsync(methodName, callback, null);
-
-        public Task SetDefaultMethodHandlerAsync(MethodCallback callback) =>
-            this.moduleClient.SetMethodDefaultHandlerAsync(callback, null);
-
-        public void Dispose() => this.moduleClient.Dispose();
+        public event EventHandler Closed;
 
-        public Task<Twin> GetTwinAsync() => this.moduleClient.GetTwinAsync();
+        public bool IsActive => this.isActive.Get();
 
-        public Task UpdateReportedPropertiesAsync(TwinCollection reportedProperties)
-            => this.moduleClient.UpdateReportedPropertiesAsync(reportedProperties);
-
-        public Task<DeviceStreamRequest> WaitForDeviceStreamRequestAsync(CancellationToken cancellationToken)
-            => this.moduleClient.WaitForDeviceStreamRequestAsync(cancellationToken);
-
-        public async Task<IClientWebSocket> AcceptDeviceStreamingRequestAndConnect(DeviceStreamRequest deviceStreamRequest, CancellationToken cancellationToken)
+        public async Task SetDesiredPropertyUpdateCallbackAsync(DesiredPropertyUpdateCallback onDesiredPropertyChanged)
         {
-            await this.moduleClient.AcceptDeviceStreamRequestAsync(deviceStreamRequest, cancellationToken);
-            return await EdgeClientWebSocket.Connect(deviceStreamRequest.Url, deviceStreamRequest.AuthorizationToken, cancellationToken);
+            try
+            {
+                this.inactivityTimer.Reset();
```

Comment:
The idea is that we create an `inner` object to do the work and reset the inactivity call on each call?

Reply:
Right. If no call is made, the inactivity timer is triggered and we close the connection.